internal/util/update/update_test.go

```diff
@@ -789,6 +790,210 @@ func TestCommand_Run_badStrategy(t *testing.T) {
+						WithKptfile().
+						WithSubPackages(
+							dataset.NewPackage("nestedbar"),
+						),
+					dataset.NewPackage("zork"),
+				),
+		},
+		{
+			name: "local updates remain after noop update",
+			initialUpstream: dataset.NewPackage("foo").
+				WithKptfile().
+				WithSubPackages(
+					dataset.NewPackage("bar").
+						WithKptfile(),
+				),
+			updatedLocal: dataset.NewPackage("foo").
+				WithKptfile().
+				WithSubPackages(
+					dataset.NewPackage("bar").
+						WithKptfile(),
+					dataset.NewPackage("zork").
+						WithKptfile(),
+				),
+			expectedLocal: dataset.NewPackage("foo").
+				WithKptfile().
+				WithSubPackages(
+					dataset.NewPackage("bar").
+						WithKptfile(),
+					dataset.NewPackage("zork").
+						WithKptfile(),
+				),
+		},
+		{
+			name: "non-overlapping additions in both upstream and local is ok",
+			initialUpstream: dataset.NewPackage("foo").
+				WithKptfile().
+				WithSubPackages(
+					dataset.NewPackage("bar").
+						WithKptfile(),
+				),
+			updatedUpstream: dataset.NewPackage("foo").
+				WithKptfile().
+				WithSubPackages(
+					dataset.NewPackage("bar").
+						WithKptfile(),
+					dataset.NewPackage("zork").
+						WithKptfile(),
+				),
+			updatedLocal: dataset.NewPackage("foo").
+				WithKptfile().
+				WithSubPackages(
+					dataset.NewPackage("bar").
+						WithKptfile(),
+					dataset.NewPackage("abc").
+						WithKptfile(),
+				),
+			expectedLocal: dataset.NewPackage("foo").
+				WithKptfile().
+				WithSubPackages(
+					dataset.NewPackage("bar").
+						WithKptfile(),
+					dataset.NewPackage("zork"),
+					dataset.NewPackage("abc").
+						WithKptfile(),
+				),
+		},
+		{
+			// TODO(mortent): This probably shouldn't work.
+			name: "overlapping additions in both upstream and local is not ok",
+			initialUpstream: dataset.NewPackage("foo").
+				WithKptfile().
+				WithSubPackages(
+					dataset.NewPackage("bar").
+						WithKptfile(),
+				),
+			updatedUpstream: dataset.NewPackage("foo").
+				WithKptfile().
+				WithSubPackages(
+					dataset.NewPackage("bar").
+						WithKptfile(),
+					dataset.NewPackage("abc").
+						WithKptfile(),
+				),
+			updatedLocal: dataset.NewPackage("foo").
+				WithKptfile().
+				WithSubPackages(
+					dataset.NewPackage("bar").
+						WithKptfile(),
+					dataset.NewPackage("abc").
+						WithKptfile(),
+				),
+			expectedLocal: dataset.NewPackage("foo").
+				WithKptfile().
+				WithSubPackages(
+					dataset.NewPackage("bar").
+						WithKptfile(),
+					dataset.NewPackage("abc").
+						WithKptfile(),
+				),
+		},
+		{
+			// TODO(mortent): It seems like the behavior here is not correct.
+			name: "subpackages deleted in upstream are deleted in fork",
+			initialUpstream: dataset.NewPackage("foo").
+				WithKptfile().
+				WithSubPackages(
+					dataset.NewPackage("bar").
+						WithKptfile(),
+				),
+			updatedUpstream: dataset.NewPackage("foo").
+				WithKptfile(),
+			expectedLocal: dataset.NewPackage("foo").
```

Comment:
This should delete `bar` package on local.

Reply:
Yeah it should, but currently it doesn't